### PR TITLE
FEATURE: support a description attribute on form template fields

### PR DESCRIPTION
**Pull request by renato**
Adds support to a new form template field attribute, `description`:

```yaml
- type: input
  id: name
  attributes:
    label: "Full name"
    placeholder: "eg. John Smith"
    description: "What is your full name?"
- type: textarea
  id: introduction
  attributes:
    label: "Introduction"
    placeholder: "A short introduction"
    description: "Write a short introdution about yourself"
```

fields with descriptions

The `Wrapper` component was also converted to a single `gjs` file.

### CODE REVIEW

File: app/assets/javascripts/discourse/tests/integration/components/form-template-field/wrapper-test.js [hunk 1]
```diff
@@ -89,7 +89,7 @@ module(
             id: 1,
             name: "Bug Reports",
             template:
-              '- type: checkbox\n  choices:\n    - "Option 1"\n    - "Option 2"\n    - "Option 3"\n  attributes:\n    label: "Enter question here"\n    description: "Enter description here"\n    validations:\n      required: true',
+              '- type: checkbox\n  id: options\n  choices:\n    - "Option 1"\n    - "Option 2"\n    - "Option 3"\n  attributes:\n    label: "Enter question here"\n    description: "Enter description here"\n    validations:\n      required: true',
```
Comment (pmusaraj) [inline app/assets/javascripts/discourse/tests/integration/components/form-template-field/wrapper-test.js:92]: Curious, is this change needed?

File: spec/lib/validators/form_template_yaml_validator_spec.rb [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+  end
+
+  describe "#check_descriptions_html" do
+    context "when description field has safe HTML" do
+      let(:yaml_content) { <<~YAML }
+          - type: input
+            id: name
+            attributes:
+              label: "Full name"
+              description: "What is your full name? Details <a href='https://test.com'>here</a>."
```
Comment (pmusaraj) [inline spec/lib/validators/form_template_yaml_validator_spec.rb:116]: Is valid HTML output in the browser? I guess we might want that, for example with links to more instructions. 

Can a rogue admin/moderator post valid HTML that breaks the layout here? (For example, by adding a `<div>` element with a class name that has complex stying in core.)

**Review by pmusaraj — COMMENTED** [review]
Can we add an acceptance or system test for the description field? We don't have to test it working on every field type, but seems reasonable to have it tested in one or two cases.

File: app/assets/javascripts/discourse/tests/integration/components/form-template-field/wrapper-test.js (hunk 1 above)
Comment (renato) [reply app/assets/javascripts/discourse/tests/integration/components/form-template-field/wrapper-test.js:92]: There's an Ember assertion on the `get` helper and it fails the test when the id is empty [here](https://github.com/discourse/discourse/blob/72b2b9238671d5520662ce9395ea183ef26958f6/app/assets/javascripts/discourse/app/components/form-template-field/wrapper.gjs#L36), because `The key provided to get must be a string or number, you passed undefined`. As ID is a mandatory field, I thought it was enough to make tests reflect this premise, but I can do some extra validation if you believe it to be helpful.

File: spec/lib/validators/form_template_yaml_validator_spec.rb (hunk 2 above)
Comment (renato) [reply spec/lib/validators/form_template_yaml_validator_spec.rb:116]: Yes, valid HTML is output. We're validating the HTML using Loofah, their SafeList is here:

https://github.com/flavorjones/loofah/blob/main/lib/loofah/html5/safelist.rb

It is kind of loose, allowing `width` or `font-size` CSS or an `h1` element, but it doesn't allow changing the `position` or complex CSS for example. I'm open to suggestions here, we can have a reduced `SafeList` with just what we want to allow (a link, essentially) or I'd be happy to use another approach entirely too, maybe there's already something in the codebase that I could reuse

Comment (renato) [reply spec/lib/validators/form_template_yaml_validator_spec.rb:116]: @pmusaraj Thank you for the review! I've added a couple of tests covering the presence of a label and description.

About the HTML description, we also have the `sanitize` gem available, and we could do something really strict like:

```ruby
    sanitized_html =
      Sanitize.fragment(description, elements: ["a"], attributes: { "a" => %w[href target] })

    is_safe_html = sanitized_html == Loofah.html5_fragment(description).to_s
```

This comparison is not ideal, but I'm unsure how to avoid it. At the same time, I wanted to avoid just saving the sanitized HTML and failing silently without a warning to the admin user - hence this validation.

Comment (pmusaraj) [reply spec/lib/validators/form_template_yaml_validator_spec.rb:116]: Yeah, I like the sanitized string option, allowing only `a` elements with a `href` and/or `target` makes a lot of sense. I think it's best to start strict and safe here. 

With the correct documentation (i.e. "you can use plain text and links for the description fields...") I'm ok with no errors being output to the user. 

Thanks for the detailed explanation!

**Review by pmusaraj — APPROVED** [review]
Thanks!
